src/index.js

```diff
@@ -64,6 +63,30 @@ function displaytask() {
       }
       localStorage.setItem('baseData', JSON.stringify(dataBase));
     });
+
+    span.addEventListener('change', () => {
+      localStorage.setItem('baseData', JSON.stringify(dataBase));
+    });
+
+    checkbox.addEventListener('change', () => {
+      if (checkbox.checked) {
+        dataBase[i].completed = true;
+      } else {
+        dataBase[i].completed = false;
+      }
+      localStorage.setItem('baseData', JSON.stringify(dataBase));
+    });
+
+    clearButton.addEventListener('click', () => {
+      if (checkbox.checked || this.completed) {
+        dataBase.splice(i, 1);
+        list.remove();
+      }
+      for (let i = 0; i < dataBase.length; i += 1) {
```

Comment:
- [ ] This functionality works until the page is refreshed. Please make sure the updated array of items is stored in local storage, so that the user gets the updated list of tasks after the page reloads.